src/libraries/System.Net.Quic/src/System.Net.Quic.csproj

```diff
@@ -142,17 +128,54 @@
+                      Version="$(SystemNetMsQuicTransportVersion)"
+                      PrivateAssets="all"
+                      Condition="'$(UseQuicTransportPackage)' == 'true'"
+                      GeneratePathProperty="true" />
+
+    <PackageReference Include="Microsoft.Native.Quic.MsQuic.Schannel"
+                      Version="$(MicrosoftNativeQuicMsQuicVersion)"
+                      PrivateAssets="all"
+                      Condition="'$(UseQuicTransportPackage)' != 'true'"
+                      GeneratePathProperty="true" />
+
     <BinPlaceDir Include="$(MicrosoftNetCoreAppRuntimePackNativeDir)" ItemName="NativeBinPlaceItem" />
     <BinPlaceDir Include="$(NetCoreAppCurrentTestHostSharedFrameworkPath)" ItemName="NativeBinPlaceItem" />
     <BinPlaceDir Include="$(LibrariesAllBinArtifactsPath)" ItemName="NativeBinPlaceItem" />
     <BinPlaceDir Include="$(LibrariesNativeArtifactsPath)" ItemName="NativeBinPlaceItem" />
-    <NativeBinPlaceItem Include="$(PkgSystem_Net_MsQuic_Transport)\runtimes\win10-$(TargetArchitecture)\native\*"
-                        Condition="'$(UseQuicTransportPackage)' == 'true'" />
     <NativeBinPlaceItem Include="$(PkgMicrosoft_Native_Quic_MsQuic_Schannel)\build\native\bin\$(TargetArchitecture)\*"
                         Condition="'$(UseQuicTransportPackage)' != 'true'" />
+    <NativeBinPlaceItem Include="$(PkgRuntime_win-x64_runtime_native_System_Net_MsQuic_Transport)\runtimes\win-$(TargetArchitecture)\native\*"
+                        Condition="'$(UseQuicTransportPackage)' == 'true'" />
+    <NativeBinPlaceItem Include="$(PkgRuntime_win-x86_runtime_native_System_Net_MsQuic_Transport)\runtimes\win-$(TargetArchitecture)\native\*"
+                        Condition="'$(UseQuicTransportPackage)' == 'true'" />
+    <NativeBinPlaceItem Include="$(PkgRuntime_win-arm64_runtime_native_System_Net_MsQuic_Transport)\runtimes\win-$(TargetArchitecture)\native\*"
+                        Condition="'$(UseQuicTransportPackage)' == 'true'" />
   </ItemGroup>
 
+  <!-- Include msquic in debug version for macOS so we can test it without packages. -->
+  <ItemGroup Condition="'$(TargetPlatformIdentifier)' == 'osx' and
+                          '$(TargetOS)' == 'osx' and
+                          '$(TargetArchitecture)' == 'x64' and
+                          '$(Configuration)' == 'Debug' and
+                          '$(DotNetBuildFromSource)' != 'true'">
+      <!-- MsQuic packages do not have macOS binaries. Our transport package is only one source. -->
+      <PackageReference Include="runtime.osx-$(TargetArchitecture).runtime.native.System.Net.MsQuic.Transport"
+                        Version="$(SystemNetMsQuicTransportVersion)"
+                        PrivateAssets="all"
+                        GeneratePathProperty="true" />
+      <BinPlaceDir Include="$(MicrosoftNetCoreAppRuntimePackNativeDir)" ItemName="NativeBinPlaceItem" />
```

Comment:
To avoid duplication, pull these out of the two ItemGroups and add single itemgroup as follows:
```xml
<ItemGroup Condition="'@(NativeBinplaceItem)' != ''">
    <BinPlaceDir Include="$(MicrosoftNetCoreAppRuntimePackNativeDir)" ItemName="NativeBinPlaceItem" />
    <BinPlaceDir Include="$(NetCoreAppCurrentTestHostSharedFrameworkPath)" ItemName="NativeBinPlaceItem" />
    <BinPlaceDir Include="$(LibrariesAllBinArtifactsPath)" ItemName="NativeBinPlaceItem" />
    <BinPlaceDir Include="$(LibrariesNativeArtifactsPath)" ItemName="NativeBinPlaceItem" />
</ItemGroup>
```

Reply:
updated.